src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -258,6 +260,15 @@ export class TerminalService implements ITerminalService {
 				? this._localTerminalsInitPromise = this._reconnectToLocalTerminals()
 				: Promise.resolve();
 		this._primaryOffProcessTerminalService = !!this._environmentService.remoteAuthority ? this._remoteTerminalService : this._localTerminalService;
+		this._primaryOffProcessTerminalService.onDidRequestDetach(async (e) => {
+			if (e.workspaceId && this._workspaceContextService.getWorkspace().id === e.workspaceId) {
+				const instanceToDetach = this.instances.find(i => i.instanceId === e.instanceId);
+				if (instanceToDetach?.persistentProcessId) {
+					this._primaryOffProcessTerminalService?.setOrphanToAttach(instanceToDetach.persistentProcessId);
```

Comment:
We'll want to use an associated request ID here, like resolve variables request id:

https://github.com/microsoft/vscode/blob/b0cde4e27114c229b385c5fa3a2b9d6ed09d8d84/src/vs/platform/terminal/node/ptyHostService.ts#L315-L315

This should help it work if 2+ terminals are dragged over at the same time.